### PR TITLE
Added sorting options

### PR DESCRIPTION
**Pull request by OlexG**

### CODE REVIEW

**Comment by OlexG** [comment]
Here is the video of the UI changes

https://user-images.githubusercontent.com/36348190/115785155-f3b19480-a373-11eb-9c6b-61c29f3789d3.mp4

File: server/controllers/postController.ts [hunk 1]
```diff
@@ -12,7 +12,12 @@ async function postPosts(req, res, next) {
 
 async function getPosts(req, res, next) {
 	if ('number' in req.query && 'page' in req.query) {
-		const result = await postManager.getPostsPage(req.query.number, req.query.page);
+		let result;
+		if ('sort' in req.query) {
+			result = await postManager.getPostsPage(req.query.number, req.query.page, req.query.sort);
+		} else {
+			result = await postManager.getPostsPage(req.query.number, req.query.page, 'default');
+		}
```
Comment (irenlian) [inline server/controllers/postController.ts]: You don't need this check. Just set the default value.
```suggestion
		const result = await postManager.getPostsPage(req.query.number, req.query.page, req.query.sort);
```

File: server/db/dbManager.ts [hunk 2]
```diff
@@ -35,7 +35,7 @@ export class PostManager {
 		return this.model.countDocuments().exec();
 	}
 
-	async getPostsPage(pageSize: number | string, pageNum: number | string): Promise<models.IPost[]> {
+	async getPostsPage(pageSize: number | string, pageNum: number | string, sort: 'default' | 'recent' | 'most-upvotes' | 'oldest'): Promise<models.IPost[]> {
```
Comment (irenlian) [inline server/db/dbManager.ts]: Use the enum here and just set the default value.
```suggestion
         enum SortOption {
             DEFAULT,
             RECENT,
             ....
         }
	async getPostsPage(pageSize: number | string, pageNum: number | string, sort: SortOption = SortOption.DEFAULT): Promise<models.IPost[]> {
```

File: server/db/dbManager.ts [hunk 3]
```diff
@@ -45,7 +45,17 @@ export class PostManager {
 		if (pageSize * (pageNum - 1) > count) {
 			return [];
 		}
-		return this.model.find().skip(pageSize * (pageNum - 1)).limit(pageSize).exec();
+		let sorted;
+		if (sort === 'default') {
```
Comment (irenlian) [inline server/db/dbManager.ts]: So here you may use `switch` for enum.

File: server/controllers/postController.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ async function postPosts(req, res, next) {
 
 async function getPosts(req, res, next) {
 	if ('number' in req.query && 'page' in req.query) {
-		const result = await postManager.getPostsPage(req.query.number, req.query.page);
+		const result = await postManager.getPostsPage(req.query.number, req.query.page, req.query.sort);
```
Comment (irenlian) [inline server/controllers/postController.ts]: It's not critical, but I'd suggest to use in these cases destruction of even rest parameters in some cases.

File: server/db/dbManager.ts [hunk 5]
```diff
@@ -45,7 +53,22 @@ export class PostManager {
 		if (pageSize * (pageNum - 1) > count) {
 			return [];
 		}
-		return this.model.find().skip(pageSize * (pageNum - 1)).limit(pageSize).exec();
+		let sorted;
+		switch (sort) {
```
Comment (irenlian) [inline server/db/dbManager.ts:57]: You should always include default section for `switch`. When your enum will expand you will avoid an error if you miss it here.

**Review by irenlian — APPROVED** [review]
Good job!

**Pull request merged**